### PR TITLE
Taking a look at OlmoEarth's landsat results

### PR DESCRIPTION
**Pull request by gabrieltseng**
A few changes in this PR:

- Adding OlmoEarth v1.2 (https://arxiv.org/abs/2605.20804v2) which required bumping olmoearth_pretrain_minimal to 0.0.6. 
- Different models prefer different input sizes. Currently, all datasets are resized to 224x224. Now, a model can declare an image_size attribute (OlmoEarth explicitly sets this to `null`, which means "ingest native image size". This is something we can do for other variable sized models like Galileo and AnySat. I don't know if this is a general solution (e.g. we probably do want to resize high-res images for OlmoEarth but not S2 images) 
- The default behaviour if olmoearth expects bands not present in the dataset is to zero them out. I don't think this is the best approach; when running benchmarks in Galileo and OlmoEarth we interpolated the closest bands (for all models, not just OlmoEarth). Right now, this is just implemented for OlmoEarth (landsat + S2) but I think this should be more general
- Forestnet is pre-normalized (uint between 0-255); we were using "norm stats from pretrained" (i.e. from OlmoEarth pretraining) but this assumes un-normalized inputs. IMO sweeping norm stats (either pre-trained or dataset) is going to be critical since forestnet is not the only dataset with this preprocessing (e.g. MADOS). Right now this is hackily implemented as "use dataset stats for landsat" but it should be dataset specific probably.
- The olmoearth configs seemed to set the min timesteps to 3 (and repeat images 3 times if there were less than 3 timesteps). OlmoEarth can ingest single timestep imagery so I removed this from the configs. 

Overall this recovers performance we were seeing in OlmoEarth. For the v1.2 nano model, forestnet accuracy -- when ingested as landsat data, not s2 -- goes from 0.25 (pre changes) to 0.36 (post changes) with a patch size of 8. This is close enough to the reported results in the paper (0.385 with a patch size of 4) imo. 

Also I removed the `landsat_as_s2` function which (as best as I could tell) was a solution for this specific dataset.

### CODE REVIEW

**Comment by isaaccorley** [comment]
This is amazing, I appreciate you looking into it! I'll rerun for all the olmoearth v1.2 models shortly

**Review by isaaccorley — APPROVED** [review]
